scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/simulation/SimulationResultsConverter.java

```diff
@@ -0,0 +1,105 @@
+import gov.nasa.jpl.aerie.constraints.model.DiscreteProfilePiece;
+import gov.nasa.jpl.aerie.constraints.model.LinearProfile;
+import gov.nasa.jpl.aerie.constraints.model.LinearProfilePiece;
+import gov.nasa.jpl.aerie.constraints.time.Interval;
+import gov.nasa.jpl.aerie.merlin.driver.SimulatedActivity;
+import gov.nasa.jpl.aerie.merlin.driver.SimulationResults;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+import gov.nasa.jpl.aerie.merlin.protocol.types.RealDynamics;
+import gov.nasa.jpl.aerie.merlin.protocol.types.SerializedValue;
+import gov.nasa.jpl.aerie.merlin.protocol.types.ValueSchema;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static gov.nasa.jpl.aerie.merlin.protocol.types.Duration.MICROSECONDS;
+
+public class SimulationResultsConverter {
+
+  /**
+   * convert a simulation driver SimulationResult to a constraint evaluation engine SimulationResult
+   *
+   * @param driverResults the recorded results of a simulation run from the simulation driver
+   * @return the same results rearranged to be suitable for use by the constraint evaluation engine
+   */
+  public static gov.nasa.jpl.aerie.constraints.model.SimulationResults convertToConstraintModelResults(
+      SimulationResults driverResults, Duration planDuration){
+    final var activities =  driverResults.simulatedActivities.entrySet().stream()
+                                                             .map(e -> convertToConstraintModelActivityInstance(e.getKey().id(), e.getValue(), driverResults.startTime))
+                                                             .collect(Collectors.toList());
+    return new gov.nasa.jpl.aerie.constraints.model.SimulationResults(
+        Interval.between(Duration.ZERO, planDuration),
+        activities,
+        Maps.transformValues(driverResults.realProfiles, SimulationResultsConverter::convertToConstraintModelLinearProfile),
+        Maps.transformValues(driverResults.discreteProfiles, SimulationResultsConverter::convertToConstraintModelDiscreteProfile)
+    );
+  }
+
+  /**
+   * convert an activity entry output by the simulation driver to one suitable for the constraint evaluation engine
+   *
+   * @param id the name of the activity instance
+   * @param driverActivity the completed activity instance details from a driver SimulationResult
+   * @return an activity instance suitable for a constraint model SimulationResult
+   */
+  public static gov.nasa.jpl.aerie.constraints.model.ActivityInstance convertToConstraintModelActivityInstance(
+      long id, SimulatedActivity driverActivity, final Instant startTime)
+  {
+    final var startT = Duration.of(startTime.until(driverActivity.start(), ChronoUnit.MICROS), MICROSECONDS);
+    final var endT = startT.plus(driverActivity.duration());
+    final var activityWindow = startT.isEqualTo(endT)
```

Comment:
Why was this variable renamed? It's of type `Interval`, not `Window`.

Reply:
I have renamed it to `activityInterval`.